### ISSUE TITLE
Pseudo-Backports [enhancement]

### ISSUE THREAD

**Issue by DVorriose** [Meldexun/BetterDiving#63]
**Enhancement**
TL;DR:  Looking for "waterlogged" blocks and underwater biome gen in 1.12.2.  Are you willing/able before moving to newer versions?  (asking a lot, I know).

**Versions**
Better Diving:  Latest
Forge:  Latest 
Minecraft:  1.12.2

Sometimes, I hate to see people come in and suggest things to mod authors, because it seems like they expect the author to immediately implement all their ideas.  Please understand, these are the things I've been looking at or having trouble with and I have absolutely no expectations.  BetterDiving is a wonderful mod and has made my pack making and gaming much better.  Thank you for the mod as is, and I hope you find a gem or two in here you like.

Background:  My two biggest complaints about water mechanics in 1.12.2 are lack of ability to build and lack of anything to see (or to see it).

Building:  Waterfalls are an issue.  
1. Can't see out glass.  Are you able to remove waterfalls against transparent blocks?
2. Can't build out of anything except full blocks.  Stairs, slabs, fences, all make for ugly builds.  A "true" waterlogged block probably isn't possible, but can you make a fake one?  
Other mods to look at:
3. Heat and Climate has some "waterlogged" blocks and glass that works underwater.
4. New mod "Underwater World" has some new, "waterlogged" blocks.
5. Tropicraft bamboo is waterlogged in tropicraft water.
One issue with all three, they take more item IDs.
Suggestions (not sure how difficult these are):
6. Change how water renders around existing Vanilla blocks (and prob some modded blocks using vanilla methods).
7. Introduce a "cut" feature, ala Chisels & Bits, Little Tiles, Rockhounding, or ArchitectureCraft that allows players to convert an existing block into a waterlogged shape.
8. Introduce Blockcraftery/Carpenters Blocks -like function where players can right-click a block onto another structure.
9. If nothing else, then new waterlogged blocks that might be used underwater, like slab/stair/fence/gate/door/trapdoor versions of concrete, prismarine, sandstone, and maybe a couple other materials.  If possible, please use Metadata; mods with more item IDs are harder to include in modpacks.

Biomes:  Only three water biomes in 1.12.2 limit exploration.  Examples:
1. Limited spawns:  Can (sometimes) use MultiMob or InControl! spawn modded fish (like BetterDiving Peepers or Abyssal Depths sea horses), but can't limit them to a spawn near BetterDiving Kelp or to CoralReef or ZAWA corals because they aren't biomes.
2. Limited structures:  Can use Recurrent Complex or probably ChocolateQuest Repoured to spawn underwater structures, like Quark (nether) smokers, backported bubble columns, dungeons/temples/monuments, and possibly even backported coral (if there were a good mod that added them), but there's only 3 biomes to spawn them in, so not a lot of new places to go.
Suggestions:
3. I think Coral, Kelp, and Iceberg all exist in later versions, seagrass plains and abysses would be nice.  
4. Splitting up into Intertidal (new shallow), Pelagic (ocean), Benthic (deep ocean), and Abyssal (deeper = more building height!) might help.
5. Minecraft caves and ravines deliberately (no air pockets) generating underwater would be interesting.
6. Increased ore spawns might draw people to build in the ocean, as might some mod integration (bigger oil deposits for Immersive Engineering?).

Some inspiration:
 - Reika's (1.7.10) GeoStrata mod
 - Minenautica (1.7.10)
 - Mariculture (1.7.10 / 1.13.2)
 - Heat and Climate (1.12.2)
 - Underwater World (1.12.2)
 - Aquatic Abyss (1.7.10) / Abyssal Depths (1.12.2 rewrite)
 - Tropicraft (1.12.2), great mobs, including schools of fish!
 - Wizzard's Animals (1.12.2 … mcreator, sadly, so not server friendly)
 - Fantastic Fish (1.7.10)
 - Railcraft, yes, Railcraft, whatever version, for abyssal stone.  What wonderful little dark hollow globes of stone sometimes filled with nice resources.

Finally, two things keep me from adding a mod to a pack.
1.  Item IDs.  JEIDs isn't compatible with everything, and usually when I run up on the ID limit the pack is pretty big already (and risks being unplayable, depending what mods are installed).  Mods with lots of IDs can be hard to fit, and it's one of the reasons I'll avoid one (like Quark, which is otherwise amazing).  Most mainstream tech/magic mods top off around 300, but I find my limit for non-primary mods is around 50.  I very much admire mods that make good use of metadata (like Rockhounding, which is otherwise massive).
2.  Speed.  Often, it's a result of too many AIs (like ZAWA) or very slow AIs (like Special AI).  Other times, it's over-complication (like Dynamic Stealth, which isn't just sight, but also touch, hearing, and tons of HUD stuff), or something obscure (not sure why Matter Overdrive is slow, but it's no longer being updated!).

I absolutely love how simple and fast and small BetterDiving is.  Thank you.  If you can, please keep it that way!  Appreciate the mod and all the work you've put into it.

**Comment by Meldexun** [comment]
So first off i edited your comment for easier references.
Building:
1. I don't know what you mean. It seems fine for me.
2. I could look into it and test if i come up with something that could work.
3. ^
4. ^
5. ^
6. I don't know what you mean. Why should i change that?
7. How that should fit into this mod? Seems like an improvement for the named mods.
8. You mean add a Carpenter'sBlocks-like stair and slab which can be waterlogged. That might work out.
9. I won't add concrete, prismarine, etc. stairs or slabs or something else.

Biomes:
The problem is that in 1.12 i can't just add a new ocean biome. It will either get treated as a normal biome (birch forest or plains) and so it won't nicely fit into the world or i would have to create a custom world type (which i won't add). Also i can't add a biome that is deeper than the deep ocean biome unless i create a custom world type.

So all in all i might look into adding some sort of waterlogged blocks. The other suggestions either don't fit or are not really possible.

**Comment by DVorriose** [comment]
Thank you for the quick response.

1.  If you put glass underwater, it gives the same moving water animation as around other transparent (non-full) blocks, like stairs, slabs, fences, etc.  That makes it hard to see out from an underwater base.  You could be in the middle of a coral reef with lots of fish swimming around, but you don't get to experience it through a window.  You can turn water animations off, I think Optifine has a switch for water animation that works, but that would switch it off elsewhere too.

I'll take a screen shot and drop it here next time I can load up.

6-9 are all just ways to possibly make waterlogged blocks.  From a player perspective, Chisels&Bits, ArchitectureCraft, and Blockcraftery essentially do the same thing in different ways.  The great thing about them is they allow most modded blocks to take on new shapes, without adding a lot of new blocks.  It'd be wonderful if we could do that underwater (and I'll probably suggest the same to those teams, but pretty sure they've all moved on).  

Thanks again.

**Comment by NordicGamerFE** [comment]
I like this idea but i feel like its better as a stand alone mod

**Comment by Meldexun** [comment]
1.12 isn't worked on anymore and this is out of the scope of this mod.
